### PR TITLE
Test Sign out and dfe sign in bad response

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

We want to increase our testing code covrage. This PR adds specs to sign out path and when the omniauth payload from dfe sign in is empty.

## Changes

Specs

## Guidance

Do tests pass?

### CODE REVIEW

File: app/models/dfe_sign_in_user.rb [hunk 1]
```diff
@@ -14,13 +14,13 @@ def initialize(email:, dfe_sign_in_uid:, first_name:, last_name:, service:, id_t
...
-      "last_name" => omniauth_payload["info"]["last_name"],
+      "email" => omniauth_payload["info"]["email"].presence,
+      "dfe_sign_in_uid" => omniauth_payload["uid"].presence,
+      "first_name" => omniauth_payload["info"]["first_name"].presence,
+      "last_name" => omniauth_payload["info"]["last_name"].presence,
       "last_active_at" => Time.current,
-      "id_token" => omniauth_payload["credentials"]["id_token"],
-      "provider" => omniauth_payload["provider"],
+      "id_token" => omniauth_payload["credentials"]["id_token"].presence,
+      "provider" => omniauth_payload["provider"].presence,
```
Comment (CatalinVoineag) [inline app/models/dfe_sign_in_user.rb:23]: Having these attributes as nil will redirect the user to the sign in page with a message like You cannot access this service. In centry, there should be an error raised by dfe sign in

**Comment by CatalinVoineag** [comment]
[[Cov] As a User, I should be presented with an authentication error if the DfE Sign In response is invalid](https://trello.com/c/3mfvqtH3/186-cov-as-a-user-i-should-be-presented-with-an-authentication-error-if-the-dfe-sign-in-response-is-invalid)

**Comment by CatalinVoineag** [comment]
[[Cov] As a User, I should be able to sign out](https://trello.com/c/LTkv29qf/185-cov-as-a-user-i-should-be-able-to-sign-out)

**Review by elceebee — APPROVED** [review]

File: app/models/dfe_sign_in_user.rb (hunk 1 above)
Comment (Nitemaeric) [reply app/models/dfe_sign_in_user.rb:23]: I'm not sure I follow here. Why do we need to add the `.presence` call in again?

Comment (CatalinVoineag) [reply app/models/dfe_sign_in_user.rb:23]: It's for setting the attributes to nil if they are not returned by dfe sign in and this https://github.com/DFE-Digital/itt-mentor-services/blob/faf4e6f9a3a1944c0bfbf603bbaacc99715356bb/app/models/dfe_sign_in_user.rb#L61
Or this or this https://github.com/DFE-Digital/itt-mentor-services/blob/faf4e6f9a3a1944c0bfbf603bbaacc99715356bb/app/models/dfe_sign_in_user.rb#L68

Will return the user to the sign in page. Before, the code would just break here if the omniauth hash was not present

Comment (Nitemaeric) [reply app/models/dfe_sign_in_user.rb:23]: Hmm, I've just removed the `.presence` calls and all the tests _still_ pass. Are we sure this is necessary?

Comment (Nitemaeric) [reply app/models/dfe_sign_in_user.rb:23]: If it's necessary, can you add a test case that fails without the addition here please?

Comment (CatalinVoineag) [reply app/models/dfe_sign_in_user.rb]: Yeah, you're right, we don't need the `presence` here

**Review by Nitemaeric — APPROVED** [review]
👍

**Pull request merged**